dali/pipeline/operators/support/random/coin_flip_test.cc

```diff
@@ -0,0 +1,39 @@
+#include <dali/pipeline/data/backend.h>
+#include "dali/test/dali_operator_test.h"
+
+namespace dali {
+
+class CoinFlipV2Test : public DaliOperatorTest<int, float> {
+
+
+  int SetInputs() const override {
+    return -1;
+  }
+
+
+  std::string SetOperator() const override {
```

Comment:
This should receive a collection of opspec objects

Reply:
I intentionally didn't want to use the OpSpec. I'd like to provide here my reasons, hope this begins a little discussion about it.

How I imagined combining GTest with our test approach is that a test fixture (by GTest definition, like `test fixture == test suite`) is a test of a specific operator(s), e.g. `[Crop]`, `[Crop -> Resize]`, `[Resize -> Crop -> Rotate]`, etc. This way we would have our requirement about testing sequence of operators fulfilled. On top of that, every test fixture here has a given set of inputs.
Furthermore, for every test fixture, it's possible to define multiple (`TEST_F`) test cases. Each of the test cases perform test using given arguments for every input.
On the other hand, OpSpec object contains information about, i.a.: Inputs, Outputs, Arguments, Operator name. Thus, combining with my approach, it OpSpec has to be created after specifying Arguments. Which are specified in `TEST_F`. And `TEST_F` is called only after instantiating fixture class.

Please, let me know, if my explanation isn't clear, I'll make some drawing or sth

Reply:
I think we all agreed to the following design:
![image](https://user-images.githubusercontent.com/39967756/49152072-fd08b500-f311-11e8-9148-885f38391fc1.png)
If you think it needs to be reconsidered, let discuss it again.

Reply:
I already discussed the design with @pribalta. Pablo, could you leave here a comment, with your opinion on my design?